Pirates are split into three different ghastly crews!

tralezab opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

Pirates are split into three different kinds of pirates:
Human pirates, who took what the skeleton pirate's old ship was. Their getup is pretty regular, they're run of the mill space pirates.
Silverscale elite royalty, who are in a giant silver shuttle. They are all snobby rich nobles hunting for sport, and if in danger they can harden into a true silver statue! Mostly, though, they use it to get the drop on you.
And our OG skeleton pirates, which have upgraded into THE FLYING DUTCHMAN. It sails the solar seas, scuppering anyone unwise enough to submit to it's bounty via giant cannons that blast people to smithereens.

## Why It's Good For The Game

Pirates have a modern kinda ftl pirate like ship but are old shanty skeletons. It didn't make sense thematically but now we're all on the same page with different types of pirates who want to extort the station.

## Changelog
:cl:
add: Pirates are now split into different variants that are all very deadly.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Krysonism committed:
cannon 1

tralezab committed:
uuuuuh pirates

tralezab committed:
a LOT of ideas, some good some bad

tralezab committed:
finished up lizard related stuff. Their powers work, their outfits are done, etc.

Qustinnus commented:
no ethereal pirates smh

MrDoomBringer commented on code/modules/shuttle/special.dm:
```diff
@@ -405,3 +405,29 @@
+	icon_state_on = "wabbajack_statue_on"
+	projectile_sound = 'sound/effects/meteorimpact.ogg'
+	active = FALSE
+	allow_switch_interact = FALSE
```
i assume ur gonna link these to buttons in mapping

tralezab replied:
Actually, kryson made cannons in the past, I'm going to use his cannons. This should be GUTTED!

MrDoomBringer commented on code/modules/surgery/organs/tongue.dm:
```diff
@@ -92,6 +92,54 @@
+		becoming_statue.visible_message("<span class='notice'>[becoming_statue] hardens into a silver statue.</span>", "<span class='notice'>You have become a silver statue!</span>")
+		statue.set_visuals(becoming_statue.appearance)
+		statue.forceMove(get_turf(becoming_statue))
+		becoming_statue.forceMove(statue)
```
players will be able to use sepia cameras and the like to escape from inside their statue form, which might cause weird behavior

tralezab replied:
I talked about this on discord, isn't this an issue with nearly all things that don't want you to actually leave? Well, I'm gonna hook signals up to the statue anyways, so..

MrDoomBringer reviewed (changes requested):
This is a draft, but I'll still point some things out that you'll wanna take a look at if you haven't already

KathrinBailey commented:
Might wanna add some scrubbers into the old pirate cutter if it's going to be crewed by things that need to breathe. That place gets a bit lung-unfriendly rather fast.

ATH1909 commented on code/modules/surgery/organs/tongue.dm:
```diff
@@ -92,6 +92,54 @@
+
+/datum/action/item_action/organ_action/statue
+	name = "Become Statue"
+	desc = "Become an elegant silver statue. It's durability and yours are directly tied together, so make sure you're careful."
```
```suggestion
	desc = "Become an elegant silver statue. Its durability and yours are directly tied together, so make sure you're careful."
```

tralezab replied:
DAMN

wesoda25 commented:
Not sure if you plan to do more work on the ships, but if not, could we see what they look like in game?

tralezab commented on code/modules/surgery/organs/tongue.dm:
```diff
@@ -92,6 +92,54 @@
+		return
+	var/is_statue = becoming_statue.loc == statue
+	to_chat(becoming_statue, "<span class='notice'>You begin to [is_statue ? "break free from the statue" : "make a glorious pose as you become a statue"]!</span>")
+	if(!do_after(becoming_statue, 5 SECONDS, target = statue))
```
oh my GOSH thank you

tralezab commented on code/modules/mob/living/carbon/human/species_types/lizardpeople.dm:
```diff
@@ -111,3 +111,37 @@ Lizard subspecies: ASHWALKERS
+/*
+Lizard subspecies: SILVER SCALED
+*/
+/datum/species/lizard/silverscale
```
Yeah actually I do :)

tralezab commented:
I do plan on doing more work so I think it's a waste to show screenies. for the moment

tralezab commented on code/modules/surgery/organs/tongue.dm:
```diff
@@ -92,6 +92,54 @@
+	if(statue.name == initial(statue.name)) //statue has not been set up
+		statue.name = "statue of [becoming_statue.real_name]"
+		statue.desc = "statue depicting [becoming_statue.real_name]"
+		statue.set_custom_materials(list(/datum/material/silver=MINERAL_MATERIAL_AMOUNT*5))
```
Yeah but silver doesn't DO that right

ATH1909 replied:
lowkey, this ability still isn't gonna be used (more than once to test it out), even if the statue could reflect lasers- it's much worse than just entering a locker and closing it on yourself, due to the long wind-up times the ability demands before use and the long cooldowns it has.

now, if the statue COULD reflect lasers (but NOT bullets), turning into a statue and back was instant (without a do_after()), AND this ability had no cooldown, THEN it could be very interesting, especially combined with the inability of the silver lizards to use guns- it'd effectively be a nerfed version of sleeping carp's projectile deflection (it'd only work on lasers and beams, not all projectiles, and it'd keep you from moving (or attacking) while it's active), giving the silver lizards a unique playstyle in comparison to the other types of pirates.

ATH1909 replied:
also, check to see what happens if you try to sell these statues while someone is still inside of them.

ATH1909 replied:
> Yeah but silver doesn't DO that right

you could change the fluff and make it turn you into a glass golem (and back) if you wanted it to be much easier to code into the game/recognize that hey, maybe you shouldn't shoot lasers at this thing

MrDoomBringer replied:
i mean maybe its polished silver
if it reflects light it reflects lasers!!
also yeah the main reason is because 5 egun shots will insta-kill-no-revive you from full health in statue mode, and in statue mode you're kinda a sitting duck. I'd say either drastically reduce the cooldown (so it can be used to quickly enter in stealth situations) or give it some kind of armor buff like reflecting lasers (to be useful situationally in combat while still requiring quick thinking/decision making)

tralezab replied:
Will lower CD

tralezab commented on code/modules/mob/living/carbon/human/species_types/lizardpeople.dm:
```diff
@@ -111,3 +111,37 @@ Lizard subspecies: ASHWALKERS
+	name = "Silver Scale"
+	id = "silverlizard"
+	limbs_id = "lizard"
+	inherent_traits = list(TRAIT_ADVANCEDTOOLUSER,TRAIT_CHUNKYFINGERS,TRAIT_NOBREATH)
```
oop

Zytolg commented:
So wait, Lizards are a Monarchy? Based.

tralezab committed:
Silverscale touch up, console has all types + admin selection

tralezab committed:
it is quicker to un-statue than it is to re-statue

ATH1909 commented on code/modules/surgery/organs/tongue.dm:
```diff
@@ -92,6 +92,54 @@
+
+/datum/action/item_action/organ_action/statue
+	name = "Become Statue"
+	desc = "Become an elegant silver statue. Its durability and yours are directly tied together, so make sure you're careful."
```
... are the statue's durability and your durability actually tied together? is that handled automatically by petrification code? what happens if you transform back while the statue is merely damaged, not destroyed?

ATH1909 commented on code/modules/surgery/organs/tongue.dm:
```diff
@@ -92,6 +92,54 @@
 		message = lizard_eckS.Replace(message, "ECKS$1")
 	speech_args[SPEECH_MESSAGE] = message
 
+/obj/item/organ/tongue/lizard/silver
```
I think these tongues might be able to be generated by meaty ores.

I personally think that that's okay, but I figured that I should at least tell you about that.

ATH1909 replied:
also, because they're tongue, you could probably farm for these by giving a silver scale the HARS mutatioin, curing it, healing them up, and then repeating the process.

you'd still have to do the tongue replacement surgeries, of course, so I think that's fine, but again, I thought that you should be made aware of this.

ATH1909 commented on code/modules/mob/living/carbon/human/species_types/lizardpeople.dm:
```diff
@@ -111,3 +111,35 @@ Lizard subspecies: ASHWALKERS
+/*
+Lizard subspecies: SILVER SCALED
+*/
+/datum/species/lizard/silverscale
```
by the way, because these are subtypes of lizards (but not ash walkers for some reason? are they supposed to not have digitigrade feet?), they're a valid magic mirror race.

I personally think that that's fine (especially if you give them TRAIT_CHUNKYFINGERS back), but hey, just so ya know.

Fikou replied:
why would they be ash walkers they are literally the opposite of them, a secret society royalty

ATH1909 replied:
fair enough

Misdoubtful commented:
Diverse midround the real MVP.

tralezab committed:
please don't run into a bug

tralezab committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into cannons-into-pirates

tralezab committed:
Merge branch 'cannons-into-pirates' into pirates-of-all-kinds

tralezab committed:
cannons are now limited by reagents, statues now clean up if the user escapes and kill user if destroyed

tralezab committed:
cutlasses for dutchman pirates

tralezab committed:
oops, name mistake

tralezab committed:
pirates.json sorted and nice

Fikou commented on code/modules/mob/living/carbon/human/species_types/lizardpeople.dm:
```diff
@@ -111,3 +111,35 @@ Lizard subspecies: ASHWALKERS
+/*
+Lizard subspecies: SILVER SCALED
+*/
+/datum/species/lizard/silverscale
```
i think they should have natural armor like golems with their silver scales, since the other pirates have armored clothing but these guys dont

tralezab replied:
They do, it's just a tad light.

Fikou replied:
also limit changesource_flags to MIRROR_BADMIN | WABBAJACK i think

tralezab committed:
dutchman has a stationary and mobile port, silverscales restricted and given light armor

ATH1909 commented on code/modules/mob/living/carbon/human/species_types/lizardpeople.dm:
```diff
@@ -111,3 +111,38 @@ Lizard subspecies: ASHWALKERS
+	species_language_holder = /datum/language_holder/lizard/silver
+	mutanttongue = /obj/item/organ/tongue/lizard/silver
+	armor = 10 //very light silvery scales soften blows
+	changesource_flags = MIRROR_BADMIN | MIRROR_MAGIC | RACE_SWAP | ERT_SPAWN
```
let people choose this from the pride mirror too

tralezab replied:
No thanks

ATH1909 commented on code/modules/surgery/organs/tongue.dm:
```diff
@@ -92,6 +92,84 @@
+		to_chat(owner, "<span class='warning'>Your body rejects the powers of the tongue!</span>")
+		return
+	var/mob/living/carbon/becoming_statue = owner
+	if(becoming_statue.health < 1)
```
what if the statue was reduced to 0.5 health before you turned back? would you just permanently be unable to turn into a statue again?

tralezab replied:
We talked on discord, not an issue

tralezab committed:
json fixed, map additions, only dutchmen enabled for testing, statue CD lowered

tralezab committed:
jesus, I am stupid/

tralezab committed:
LINTING. LINTING. LINTING.

tralezab committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into pirates-of-all-kinds

tralezab committed:
CONFLICTS. CONFLICTS. CONFLICTS.

TheVekter commented:
We test merged this on Manuel last night, the Dutchman is so _fucking cool_

Fikou commented on code/game/objects/items/weaponry.dm:
```diff
@@ -266,6 +266,20 @@ for further reading, please see: https://github.com/tgstation/tgstation/pull/301
 	ADD_TRAIT(src, TRAIT_NODROP, CURSED_ITEM_TRAIT)
 
+//statistically similar to e-cutlasses
+/obj/item/katana/cutlass
```
this seems like a weird thing to subtype off, katanas have some snowflake stuff like being able to fit in fedoras, id just make it its own thing or subtype off a claymore or something

Fikou commented on strings/pirates.json:
```diff
@@ -1,31 +1,101 @@
-}
\ No newline at end of file
+	"beginnings":[
+		"Comet",
```
```suggestion
		"Comet",
		"Dread",
```

Fikou commented on strings/pirates.json:
```diff
@@ -1,31 +1,101 @@
+		"hull",
+		"keel",
+		"knot",
+		"salt"
```
```suggestion
		"salt",
		"stump"
```

Potato-Masher commented:
I was the pirate captain for the Manuel round Vekter mentioned. And yes, the Dutchman is **VERY cool**. It was a lot of fun!

Couple of things I noticed were that the skelepirates don't come with radio headsets. Not sure if this is intentional or not, but our crew was given some headsets by the admins.

Explosive shot is perfect, but there isn't a whole lot of it that starts on the Dutchman. We used the starting explosive shot in about 10 seconds.
Round shot is a bit underwhelming. Slaps humans for decent damage and delimbs happen, but the ball just kind of vanishes after hitting someone. Maybe have it pierce structures for a certain distance like some types of meteors, and when it hits people it flings them off in a direction from the force of the impact? 
 
Cutlass doesn't have equipped sprites, but that is a minor gripe.

_Very_ funny, and a whole lot different from old skelepirates.
![wmplayer_qvN0XSeAUd](https://user-images.githubusercontent.com/33107541/105338143-37e09d80-5ba9-11eb-9a0e-102ba8adf08a.png)

tralezab committed:
The Biggest One, repaths cutlass, cutlass back sprite, outfit updates

tralezab committed:
adds malfunction shot to the cannon paper

tralezab committed:
gunpowder barrels can no longer be opened, it just makes them confusing

Fikou commented on code/modules/projectiles/projectile/bullets/cannon.dm:
```diff
@@ -0,0 +1,58 @@
+	name = "malfunction shot"
+	icon_state = "emp_cannonball"
+	projectile_piercing = NONE
+	damage = 10 //set to 30 before first mob impact, but they're gonna be gibbed by the explosion
```
comment doesn tm make sense

tralezab committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into pirates-of-all-kinds

tralezab committed:
wait, really? no conflicts? ok...

tralezab committed:
SILVERSCALE HOURS.

ATH1909 commented:
> I was the pirate captain for the Manuel round Vekter mentioned. And yes, the Dutchman is **VERY cool**. It was a lot of fun!
> 
> Couple of things I noticed were that the skelepirates don't come with radio headsets. Not sure if this is intentional or not, but our crew was given some headsets by the admins.
> 
> Explosive shot is perfect, but there isn't a whole lot of it that starts on the Dutchman. We used the starting explosive shot in about 10 seconds.
> Round shot is a bit underwhelming. Slaps humans for decent damage and delimbs happen, but the ball just kind of vanishes after hitting someone. Maybe have it pierce structures for a certain distance like some types of meteors, and when it hits people it flings them off in a direction from the force of the impact?
> 
> Cutlass doesn't have equipped sprites, but that is a minor gripe.
> 
> _Very_ funny, and a whole lot different from old skelepirates.
> ![wmplayer_qvN0XSeAUd](https://user-images.githubusercontent.com/33107541/105338143-37e09d80-5ba9-11eb-9a0e-102ba8adf08a.png)

what if we just added some sort of structure to the dutchman that dispenses cannon ammo? like, a pile of cannonballs or something

IndieanaJones commented:
I really like the skelepirates, sort of like the lizards, and really don't like the humans.  The human ship is essentially the worst of all worlds, with the boring old ship and the most boring crew out of what you've made here.  Having midround's fun factor being entirely dictated by RNG is a very lame, so I'd suggest either make them more interesting or axe them.

tralezab committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into pirates-of-all-kinds

Mothblocks committed:
Write clear rules on descriptive and obvious names in the contribution guidelines (#56872)

This has more or less been enforced already by me and Rohesie, but puts it into clear writing, as the frequency of these names stays about the same.

Formally bans variables like M, C, and H. Writes an exclusion for `i`, as though I don't like this one, I believe other maintainers disagree. Let me know if I'm misinformed.

Also adds some additional details on negated names, such as `is_not_flying` -> `is_flying`.

tralezab committed:
restores dmis

itseasytosee committed:
Gives wizard constructs a purple theme. Minor construct fixes.  (#56758)

tgstation-server committed:
Automatic changelog generation for PR #56758 [ci skip]

tralezab committed:
resolved le lint, just need to get the silverscale navigation right

tralezab committed:
silvescale shuttle is cool dog now

tralezab committed:
Revert "Automatic changelog generation for PR #56758 [ci skip]"

This reverts commit 30e4a1aea16b0d7f2f067e95c92e58606e514ad2.

tralezab committed:
Revert "Gives wizard constructs a purple theme. Minor construct fixes.  (#56758)"

This reverts commit 070b722a8a5be03681167a09b76c9bcc7fd27001.

tralezab committed:
Revert "Write clear rules on descriptive and obvious names in the contribution guidelines (#56872)"

This reverts commit 645516f0c15c05c538bfc439cee49a1d10a343e0.

tralezab committed:
dmm 2 tgm

tralezab committed:
accidentally left in old sprite

tralezab commented:
ready

stylemistake committed:
Delete tgui-common.bundle.js

stylemistake commented:
Bonk @tralezab

Mothblocks commented on code/game/objects/structures/cannon.dm:
```diff
@@ -0,0 +1,104 @@
+	create_reagents(charge_size)
+
+/obj/structure/cannon/proc/fire()
+	for(var/mob/M in urange(10, src))
```
https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names

Mothblocks commented on code/game/objects/structures/cannon.dm:
```diff
@@ -0,0 +1,104 @@
+	if(istype(W, /obj/item/stack/cannonball))
+		if(loaded_cannonball)
+			to_chat(user, "<span class='warning'>[src] is already loaded!</span>")
+			return
```
Move the return outside of the if, since they both end the same way.

Mothblocks commented on code/game/objects/structures/ghost_role_spawners.dm:
```diff
@@ -583,7 +584,8 @@
 	assignedrole = "Space Pirate"
-	var/rank = "Mate"
+	var/rank = "Deserter"
+	var/spawn_oldpod = TRUE
```
Document.

Mothblocks commented on code/game/objects/structures/ghost_role_spawners.dm:
```diff
@@ -595,16 +597,64 @@
+
+/obj/effect/mob_spawn/human/pirate/silverscale/generate_pirate_name()
+	var/first_name
+	if(gender == MALE)
```
Change to a switch/case that has one for MALE, FEMALE, and neither (in which case you'd pick from male + female).

Mothblocks commented on code/modules/clothing/outfits/standard.dm:
```diff
@@ -121,21 +121,57 @@
+
+/datum/outfit/pirate/silverscale/post_equip(mob/living/carbon/human/H)
+	..()
+	var/obj/item/card/id/W = H.wear_id
```
https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names

Mothblocks commented on code/modules/surgery/organs/tongue.dm:
```diff
@@ -92,6 +92,83 @@
+
+/datum/action/item_action/organ_action/statue/New(Target)
+	. = ..()
+	statue = new(null)
```
```suggestion
	statue = new
```

Mothblocks commented on code/modules/surgery/organs/tongue.dm:
```diff
@@ -92,6 +92,83 @@
+		to_chat(becoming_statue, "<span class='warning'>Your transformation is interrupted!</span>")
+		ability_cooldown = world.time + 3 SECONDS
+		return
+	ability_cooldown = world.time + 10 SECONDS
```
Use the cooldown defines.

Mothblocks commented on code/modules/surgery/organs/tongue.dm:
https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names

tralezab committed:
mothblocks review

Mothblocks commented on code/game/objects/structures/ghost_role_spawners.dm:
```diff
@@ -639,12 +641,15 @@
+			first_name = pick(GLOB.lizard_names_male)
+		if(FEMALE)
+			first_name = pick(GLOB.lizard_names_female)
+		if(NEUTER)
```
```suggestion
		else
```

We have plural gender too.

tralezab committed:
Update code/game/objects/structures/ghost_role_spawners.dm

Co-authored-by: Mothblocks <35135081+Mothblocks@users.noreply.github.com>

adamkad1 commented:
friggin lizards got like no guns whatsoever
